backend/src/main/scala/com/softwaremill/adopttapir/starter/content/ContentService.scala

```diff
@@ -21,5 +21,5 @@ final case class ContentService(generatedFilesFormatter: GeneratedFilesFormatter
         DirectoryMerger(projectName, paths, content)
       }))
       projectAsTree <- IO(projectAsDirTrees.reduce(DirectoryMerger.apply))
-      _ <- IO(Metrics.increasePreviewOperationMetricCounter(starterDetails))
+      _ <- Metrics.increasePreviewOperationMetricCounter(starterDetails)
```

Comment:
ha that might be the reason why the number of 200 requests didn't match the metrics - we never counted previews ;)

Reply:
good to have a test :)

Reply:
yeah I was wondering about the same :D

Reply:
BTW shouldn't this emit a compiler warning?

Reply:
yeah - see the following
```
$ sbt clean compile | grep ContentService -c
0
```

I guess that in order to achieve it we would have to return sth from it and then ignore it at level uo...

Reply:
On the bright side we have a test for it now 😄

Reply:
`-Wvalue-discard` I guess should do the trick ... we don't have it by default? ;)

Reply:
it's there (`show scalacOptions`). But that's a different case, we're not casting anything to unit, we simply created an `IO[IO[...]]` and explicitly discarded the result (`_ <- ...`). Probably uncatchable by the compiler

Reply:
Yeah, I meant exactly this. At first - "we should be able to catch this with compiler flags!". But then - "we are doing `_ <- ...`, no hope there".